sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala

```diff
@@ -964,6 +964,52 @@ class QueryCompilationErrorsSuite
         "className" -> "org.apache.spark.sql.catalyst.expressions.UnsafeRow"))
   }
 
+  test("SPARK-47102: Collate in CollateClauseContext when COLLATION_ENABLED is false") {
```

Comment:
I think you can fold all those 4 tests to one like:
```scala
  test("SPARK-47102: the collation feature is off") {
    withSQLConf(SQLConf.COLLATION_ENABLED.key -> "false") {
      Seq(
        "CREATE TABLE t(col STRING COLLATE 'UNICODE_CI') USING parquet",
        "SELECT collate('aaa', 'UNICODE_CI')",
        "SELECT 'aaa' COLLATE 'UNICODE_CI'",
        "select collation('aaa')"
      ).foreach { sqlText =>
        checkError(
          exception = intercept[AnalysisException](sql(sqlText)),
          errorClass = "UNSUPPORTED_FEATURE.COLLATION")
      }
    }
  }
```

Reply:
Some tests need context because of the change I introduced in the builder stage, so I incorporated your feedback in two separate tests.